builtin/settingtypes.txt

```diff
@@ -933,112 +1093,174 @@ movement_gravity (Gravity) float 9.81
 
 #    A chosen map seed for a new map, leave empty for random.
 #    Will be overridden when creating a new world in the main menu.
+#
+#    Requires: show_advanced
 fixed_map_seed (Fixed map seed) string
 
 #    Name of map generator to be used when creating a new world.
 #    Creating a world in the main menu will override this.
 #    Current mapgens in a highly unstable state:
 #    -    The optional floatlands of v7 (disabled by default).
+#
+#    Requires: show_advanced
 mg_name (Mapgen name) enum v7 v7,valleys,carpathian,v5,flat,fractal,singlenode,v6
 
 #    Water surface level of the world.
+#
+#    Requires: show_advanced
 water_level (Water level) int 1 -31000 31000
 
 #    From how far blocks are generated for clients, stated in mapblocks (16 nodes).
+#
+#    Requires: show_advanced
 max_block_generate_distance (Max block generate distance) int 10 1 32767
 
 #    Limit of map generation, in nodes, in all 6 directions from (0, 0, 0).
 #    Only mapchunks completely within the mapgen limit are generated.
 #    Value is stored per-world.
+#
+#    Requires: show_advanced
 mapgen_limit (Map generation limit) int 31007 0 31007
 
 #    Global map generation attributes.
 #    In Mapgen v6 the 'decorations' flag controls all decorations except trees
 #    and jungle grass, in all other mapgens this flag controls all decorations.
+#
+#    Requires: show_advanced
 mg_flags (Mapgen flags) flags caves,dungeons,light,decorations,biomes,ores caves,dungeons,light,decorations,biomes,ores,nocaves,nodungeons,nolight,nodecorations,nobiomes,noores
 
 [*Biome API noise parameters]
 
 #    Temperature variation for biomes.
+#
+#    Requires: show_advanced
 mg_biome_np_heat (Heat noise) noise_params_2d 50, 50, (1000, 1000, 1000), 5349, 3, 0.5, 2.0, eased
 
 #    Small-scale temperature variation for blending biomes on borders.
+#
+#    Requires: show_advanced
 mg_biome_np_heat_blend (Heat blend noise) noise_params_2d 0, 1.5, (8, 8, 8), 13, 2, 1.0, 2.0, eased
 
 #    Humidity variation for biomes.
+#
+#    Requires: show_advanced
 mg_biome_np_humidity (Humidity noise) noise_params_2d 50, 50, (1000, 1000, 1000), 842, 3, 0.5, 2.0, eased
 
 #    Small-scale humidity variation for blending biomes on borders.
+#
+#    Requires: show_advanced
 mg_biome_np_humidity_blend (Humidity blend noise) noise_params_2d 0, 1.5, (8, 8, 8), 90003, 2, 1.0, 2.0, eased
 
 [*Mapgen V5]
 
 #    Map generation attributes specific to Mapgen v5.
+#
+#    Requires: show_advanced
```

Comment:
This is very repetitive, especially for mapgens. Could this functionality be added for entire categories which is then propagated to contained settings?